tools/distrib/python/docgen.py

```diff
@@ -109,17 +103,16 @@
     print('Cloning your repository...')
     subprocess.check_call(
         [
-            'git', 'clone', 'https://{}@github.com/{}/grpc'.format(
-                github_user, github_repository_owner)
+            'git',
+            'clone',
+            '--depth',
+            '1',
+            '--branch',
+            'gh-pages',
+            'git@github.com:%s/grpc.git' % (github_repository_owner),
```

Comment:
It's preferable to use `https://` when checking out instead of SSH. In the past we have had issues in other scripts when they were checking out git@... due to interference with SSH agent, security keys, GitHub permissions, etc.

Reply:
Thanks for mentioning that, I didn't know that context. The `--submit` flag is used to automate the submit process, and with public key set the SSH clone should be more handy, since it doesn't ask the password or two-factor authentication every time.

Reply:
You can checkout via `https://` and submit via `git@` by adding a remote.

Reply:
Tried in docker container, the `https://` will request for
```
git config --global user.email "you@example.com"
git config --global user.name "Your Name"
```

If the the local git environment doesn't support clone, it will not able to support push for sure. And since the created clone folder was a temp folder, we can't expect the user to reuse the cloned copy to push. The optimization may only valid for user who sort of half done with git configuration...

Reply:
Are you saying the following does not work:

```
git clone https://github.com/grpc/grpc -b gh-pages .
# on submit only
git remote add grpc git@github.com:grpc/grpc
git push grpc gh-pages
```

It's strange: the complaint about `git config` should not have anything to do with whether you are cloning from http or ssh, but whether your user has a `.gitconfig` already configured. I'm pretty sure if you don't have a `.gitconfig` in your container you will get the same thing even with `git@...` URL.

Reply:
To push to GitHub, one should have their Git environment setup ready, including `.gitconfig` and public key. My intent is to make the script do the right thing in a simple way. The user of this script will be the person who does the release process, and they supposed to be able to push to GitHub. Do you have more detailed fail cases?

Reply:
Thank you for reviewing and suggesting. Yeah, this will be a problem if the user has modification on their own repository's `gh-pages` branch. And I also found that shallow clone cannot be pushed to new repo, even if it is forked  [Stackoverflow](https://stackoverflow.com/questions/28983842/remote-rejected-shallow-update-not-allowed-after-changing-git-remote-url). Fixed.

Reply:
Your solution I think is both correct and convenient ;)